types/itemsjs/index.d.ts

```diff
@@ -0,0 +1,150 @@
+// Type definitions for itemsjs 2.1
+// Project: https://github.com/itemsapi/itemsjs
+// Definitions by: Adam Thompson-Sharpe <https://github.com/MysteryBlokHed>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export as namespace itemsjs;
+
+declare namespace itemsjs {
+    interface Bucket<I extends {}> {
+        key: keyof I & string;
+        doc_count: number;
+        selected: boolean;
+    }
+
+    type Buckets<I extends {}> = Array<Bucket<I>>;
+
+    interface SearchAggregation<I extends {}, A extends string> {
+        name: A;
+        title: string;
+        position: number;
+        buckets: Buckets<I>;
+    }
+
+    interface Pagination {
+        page: number;
+        per_page: number;
+        total: number;
+    }
+
+    interface SearchOptions<I extends {}, S extends string, A extends string> {
+        query?: string | undefined;
+        /** @default 1 */
+        page?: number | undefined;
+        /** @default 12 */
+        per_page?: number | undefined;
+        /** The name of a sort defined in the configuration's sortings, or a new custom one */
+        sort?: S | Sorting<I> | undefined;
+        filters?: Partial<Record<A, string[]>> | undefined;
+        /** A custom function to filter values */
+        filter?: ((item: I) => boolean) | undefined;
+        /** @default false */
+        isExactSearch?: boolean | undefined;
+        /** @default false */
+        removeStopWordFilter?: boolean | undefined;
+        /** @default false */
+        is_all_filtered_items?: boolean | undefined;
+    }
+
+    interface AggregationOptions<A extends string> {
+        name: A;
+        /** @default 1 */
+        page?: number | undefined;
+        /** @default 10 */
+        per_page?: number | undefined;
+        query?: string | undefined;
+        conjunction?: boolean | undefined;
+    }
+
+    interface SimilarOptions<I extends {}> {
+        field: keyof I & string;
+        /** @default 0 */
+        minimum?: number | undefined;
+        /** @default 1 */
+        page?: number | undefined;
+        /** @default 10 */
+        per_page?: number | undefined;
+    }
+
+    interface itemsjs<I extends {}, S extends string, A extends string> {
```

Comment:
Please stick to common convention, this should be CamelCase, isn't it?

Reply:
Yes, my mistake. Just changed it